src/main/java/net/neoforged/neoforge/attachment/IAttachmentHolder.java

```diff
@@ -110,4 +110,28 @@ default <T> T getExistingDataOrNull(Supplier<AttachmentType<T>> type) {
     default <T> @Nullable T removeData(Supplier<AttachmentType<T>> type) {
         return removeData(type.get());
     }
+
+    /**
+     * Syncs a data attachment of the given type with all relevant clients.
+     *
+     * <p>If there is currently no attachment of the given type,
+     * the removal of the attachment is synced to the client.
+     *
+     * @see AttachmentSyncHandler
+     */
+    default void syncData(AttachmentType<?> type) {
+        // Do nothing by default, implementers should override this method if needed.
```

Comment:
Should this do nothing by default, or should it throw? If we have it throw it becomes super obvious if someone tries to sync data from an un-syncable holder.

Reply:
Throwing is quite impractical because a lot of code (in `getData`, `setData`, `removeData`) simply calls `syncData` to perform any syncing if needed.

Note that proto chunks as well as client-side holders are not syncable, and in that case it's easiest to just ignore the request to sync the data.